datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -328,6 +328,15 @@ public SdxCluster getByNameInAccount(String userCrn, String name) {
         return createSdx(userCrn, name, sdxClusterRequest, internalStackV4Request, null);
     }
 
+    public SdxCluster updateRangerRazEnabled(SdxCluster sdxCluster) {
```

Comment:
Yes, since this point will be reached only if Raz has been installed (and `rangerRazEnabled` has not been set to true), the flag needs to be switched and persisted in the DB. It's also done to verify the output of `cdp datalake describe-datalake --datalake-name "aakash-dl"` shows `enableRangerRaz` as true.

Reply:
still, this endpoint could be called, the flag set to true even if RAZ is not enabled on the cluster

Reply:
Okay. Workarounds for this can be:
- Move the functionality which checks for Raz being installed from [`ClouderaManagerClusterStatusService`](https://github.com/hortonworks/cloudbreak/blob/CB-14317/cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerClusterStatusService.java#L266), and [`StackStatusCheckerJob`](https://github.com/hortonworks/cloudbreak/blob/CB-14317/core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java#L219) to [`SdxClusterStatusCheckerJob`](https://github.com/hortonworks/cloudbreak/blob/CB-14317/datalake/src/main/java/com/sequenceiq/datalake/job/SdxClusterStatusCheckerJob.java). That way, the Sdx service layer can be directly invoked and a new API endpoint need not be introduced. (Dependency issues may be encountered)
- Create a new separate job to check different services installed on the cluster. This job can be called infrequently so that it isn't an expensive operation.

Reply:
still I don't understand why we expect that RAZ is turned on outside off the controlplane? why don't we build the feature to turn it on from the controlplane?

Reply:
Yes, the complete feature to automate raz installation is tracked in [CRB-1615](https://jira.cloudera.com/browse/CRB-1615) which is a much bigger change. But this requirement is for our customer Abbvie, where the steps in [the Raz installation runbook](https://docs.google.com/document/d/1k-wIQ8l7-UZAZ4GcG0rOfUHZdvbFpBfRLZBUUHXj7Do/edit#heading=h.s4qnzj6baf5) will be followed, so currently, it is given that Raz will be installed manually.

Reply:
so is this for a one time thing?

Reply:
@lacikaaa 
Automation of the whole workflow to enable RAZ end to end will probably take a few quarters. We will through this manual runbook for a few customers to understand the issues, and then take up automation

Reply:
I see. So I would stick with a separate RAZ sync job, which would check if RAZ is enabled in the cluster. This should be in cb core in my opinion. We already sync cluster status from cb core to datalake service, that could include this flag update logic, there shouldn't be a separate API call. That one mislead me a bit earlier.
But I will let @lnardai share his opinion on this one

Reply:
Okay, will make changes to add a job in CB core.
> We already sync cluster status from cb core to datalake service
Could you share the class that is invoked during this process?